test/extended/etcd/invariants.go

```diff
@@ -49,3 +72,59 @@ var _ = g.Describe("[sig-etcd] etcd", func() {
 		}
 	})
 })
+
+func getCurrentNetworkTopology(oc *exutil.CLI) string {
+	configClient, err := configv1client.NewForConfig(oc.AdminConfig())
+	o.Expect(err).ToNot(o.HaveOccurred())
+	nework, err := configClient.ConfigV1().Networks().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+	ipFamily, err := getPreferredIPFamily(nework)
+	o.Expect(err).ToNot(o.HaveOccurred())
+	return ipFamily
+}
+
+// GetPreferredIPFamily checks network status for service CIDR to conclude IP family. If status is not yet populated fallback to spec.
+func getPreferredIPFamily(network *configv1.Network) (string, error) {
+	var serviceCIDR string
+	switch {
+	case len(network.Status.ServiceNetwork) != 0:
+		serviceCIDR = network.Status.ServiceNetwork[0]
+		if len(serviceCIDR) == 0 {
+			return "", fmt.Errorf("networks.%s/cluster: status.serviceNetwork[0] is empty", configv1.GroupName)
+		}
+		break
+	case len(network.Spec.ServiceNetwork) != 0:
+		klog.Warningf("networks.%s/cluster: status.serviceNetwork not found falling back to spec.serviceNetwork", configv1.GroupName)
+		serviceCIDR = network.Spec.ServiceNetwork[0]
+		if len(serviceCIDR) == 0 {
+			return "", fmt.Errorf("networks.%s/cluster: spec.serviceNetwork[0] is empty", configv1.GroupName)
+		}
+		break
+	default:
+		return "", fmt.Errorf("networks.%s/cluster: status|spec.serviceNetwork not found", configv1.GroupName)
+	}
+
+	ip, _, err := net.ParseCIDR(serviceCIDR)
+
+	switch {
+	case err != nil:
+		return "", err
+	case ip.To4() == nil:
+		return "tcp6", nil
+	default:
+		return "tcp4", nil
+	}
+}
+
+func isIPv4(ipString string) (bool, error) {
```

Comment:
took it from https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/dnshelpers/util.go#L108